webapp/loggedin/surveyViews/record/actions.js

```diff
@@ -54,8 +54,9 @@ export const createRecord = (history, preview = false) => async (dispatch, getSt
   const state = getState()
   const user = AppState.getUser(state)
   const surveyId = SurveyState.getSurveyId(state)
+  const cycle = User.getPrefSurveyCycle(surveyId)(user)
 
```

Comment:
please, use ```SurveyState.getSurveyCycleKey```

Reply:
since we have to get user and surveyId from the state already, it's better to use directly the User function, isn't it?

Reply:
nope.. I would prefer one common place to get the current cycle